src/network/networkd-address.c

```diff
@@ -775,6 +779,47 @@ int config_parse_label(const char *unit,
         return 0;
 }
 
+int config_parse_lifetime(const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        Network *network = userdata;
+        _cleanup_address_free_ Address *n = NULL;
+        int r, k;
+
+        assert(filename);
+        assert(section);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = address_new_static(network, section_line, &n);
+        if (r < 0)
+                return r;
+
+        if (!streq(rvalue, "forever")) {
+                r = safe_atoi(rvalue, &k);
+                if (r < 0) {
+                        log_syntax(unit, LOG_ERR, filename, line, r, "Failed to parse PreferredLifetime, ignoring: %s", rvalue);
+                        return 0;
+                }
+
+                if (k != 0)
+                        log_syntax(unit, LOG_ERR, filename, line, 0, "Invalid PreferredLifetime, value, ignoring: %d", k);
+        }
+
+        n->preferred_lifetime = k;
```

Comment:
I don't follow that explanation, sorry :) Where is `k` initialized? And I wonder how this could have worked in tests.

Maybe we should really have integration tests for all these added features and their possible values.